src/helpers.rs

```diff
@@ -462,14 +471,51 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         fn bytes_to_os_str<'tcx, 'a>(bytes: &'a [u8]) -> InterpResult<'tcx, &'a OsStr> {
             let s = std::str::from_utf8(bytes)
                 .map_err(|_| err_unsup_format!("{:?} is not a valid utf-8 string", bytes))?;
-            Ok(&OsStr::new(s))
+            Ok(OsStr::new(s))
         }
 
         let this = self.eval_context_ref();
         let bytes = this.memory.read_c_str(scalar)?;
         bytes_to_os_str(bytes)
     }
 
+    /// Helper function to read an OsString from a 0x0000-terminated sequence of u16,
+    /// which is what the Windows APIs usually handle.
+    fn read_os_str_from_wide_str<'a>(&'a self, scalar: Scalar<Tag>) -> InterpResult<'tcx, OsString>
+    where
+        'tcx: 'a,
+        'mir: 'a,
```

Comment:
Since `'a` is not used in the return type, I do not think these `where` clauses here are necessary.